<!DOCTYPE html>
<html lang="en"><head>
<script src="day1_files/libs/clipboard/clipboard.min.js"></script>
<script src="day1_files/libs/quarto-html/tabby.min.js"></script>
<script src="day1_files/libs/quarto-html/popper.min.js"></script>
<script src="day1_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="day1_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="day1_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="day1_files/libs/quarto-html/quarto-syntax-highlighting-226bd0f977fa82dfae4534cac220d79a.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.7.23">

  <meta name="author" content="Kerry Cobb">
  <title>ATAC-seq / Chip-seq Workshop</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="day1_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="day1_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    html { -webkit-text-size-adjust: 100%; }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; font-weight: bold; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="day1_files/libs/revealjs/dist/theme/quarto-250687541b6fd0303e6fa9a425de9713.css">
  <link href="day1_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="day1_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="day1_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="day1_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
  <link href="custom.css" rel="stylesheet">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">ATAC-seq / Chip-seq Workshop</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Kerry Cobb 
</div>
</div>
</div>

</section>
<section>
<section id="objectives" class="title-slide slide level1 center">
<h1>Objectives</h1>
<ul>
<li>Understand the principles behind ATAC-seq and ChIP-seq</li>
<li>Learn about experimental workflow for ATAC-seq and ChIP-seq</li>
<li>Gain familiarity with data analysis tools for ATAC-seq and ChIP-seq</li>
<li>Have an example of for project organization</li>
<li>Understand principles of HPC usage</li>
<li>Gain experience with Linux command line</li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="format-of-workshop" class="slide level2">
<h2>Format of Workshop</h2>
<ul>
<li>Largely cookbook style</li>
<li>Some hands-on exercises</li>
<li>Learn principles behind methods</li>
<li>Please interrupt if you have questions!</li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="why-use-atac-seq-chip-seq" class="slide level2">
<h2>Why use ATAC-seq / ChIP-seq?</h2>
<ul>
<li>Understand gene regulation <!-- TODO: Put some generic image here, preferably reused --></li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="gene-regulation" class="slide level2">
<h2>Gene Regulation</h2>
<ul>
<li><span class="fragment highlight-red">Transcription Regulation</span>
<ul>
<li>When and how much a gene is transcribed</li>
</ul></li>
<li>Translation Regulation
<ul>
<li>When and how much mRNA is translated into protein</li>
</ul></li>
<li>Post-translation Regulation
<ul>
<li>Activation, inactivation, and degradation of proteins</li>
</ul></li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="regulation-of-transcription" class="slide level2">
<h2>Regulation of Transcription</h2>
<ul>
<li>How genes are turned on or off in different cells and conditions</li>
<li>Involves interactions between DNA, RNA, and proteins</li>
<li>Key players:
<ul>
<li>Epigenetic modifications</li>
<li>Chromatin structure</li>
<li>Transcription factors</li>
</ul></li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="epeigenetic-modifications" class="slide level2">
<h2>Epeigenetic Modifications</h2>

<!------------------------------------------------------------------>
<img data-src="img/methylation.png" class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="chromatin-structure" class="slide level2">
<h2>Chromatin Structure</h2>
<ul>
<li>Chromatin: The complex of DNA and proteins (mainly histones) that make up chromosomes</li>
</ul>

<!------------------------------------------------------------------>
<img data-src="img/chromatin-organization_jeffrey-hansen.jpg" class="r-stretch quarto-figure-center"><p class="caption">Jeffrey Hansen</p></section>
<section id="transcription-factors" class="slide level2">
<h2>Transcription Factors</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/transcription-factors.png" class="quarto-figure quarto-figure-center" width="800"></p>
</figure>
</div>
<!------------------------------------------------------------------>
</section>
<section id="transcription-factors-1" class="slide level2 smaller">
<h2>Transcription factors</h2>
<ul>
<li>A protein that binds to specific DNA sequences to regulate the transcription of genetic information from DNA to messenger RNA</li>
<li>Contains at least one DNA-binding domain (DBD) that recognizes specific DNA sequences adjacent to the genes they regulate</li>
<li>Types
<ul>
<li>Activator - A transcription factor binding to an enhancer</li>
<li>Repressor - A transcription factor binding to a silencer</li>
<li>Coactivator - A protein that binds to an activator and helps to increase transcription</li>
</ul></li>
<li>2,800 proteins in human genome have DNA binding domains!</li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="focus-of-workshop" class="slide level2">
<h2>Focus of Workshop</h2>
<ul>
<li>DNA Accessibility
<ul>
<li>Primarily studied using ATAC-seq</li>
<li>ChIP-seq targeting H3K27ac
<ul>
<li>Epigenetic modification marking active gene regulatory regions</li>
</ul></li>
</ul></li>
<li>Transcription Factor Binding
<ul>
<li>Primarily studied using ChIP-seq</li>
<li>Bound transcription factors leave signature in ATAC-seq data as well</li>
</ul></li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="atac-seq" class="slide level2">
<h2>ATAC-seq</h2>
<ul>
<li>Assay for Transposase-Accessible Chromatin</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/atac-seq_stefan-dillinger.png" class="" width="600"></p>
<figcaption>Stefan Dillinger</figcaption>
</figure>
</div>
<!------------------------------------------------------------------>
</section>
<section id="atac-seq-1" class="slide level2">
<h2>ATAC-seq</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/atac-library_fiorella-grandi.png" class="" width="500"></p>
<figcaption>Fiorella Grandi</figcaption>
</figure>
</div>
</section>
<section id="atac-seq-2" class="slide level2">
<h2>ATAC-seq</h2>
<p><img data-src="img/atac-library_fiorella-grandi.png" class="nostretch" width="500"> <img data-src="img/atac-library-legend.png" class="nostretch" width="400" alt="Fiorella Grandi"></p>
<!------------------------------------------------------------------>
</section>
<section id="illumina-sequencing" class="slide level2">
<h2>Illumina Sequencing</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/illumina-sequencing_ahmed-shuikan.png" class="" width="800"></p>
<figcaption>Ahmed Shuikan</figcaption>
</figure>
</div>
</section>
<section id="illumina-sequencing-1" class="slide level2">
<h2>Illumina Sequencing</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/sequencing-by-synthesis.jpg" class="quarto-figure quarto-figure-center" width="800"></p>
</figure>
</div>
<!------------------------------------------------------------------>
</section>
<section id="atac-seq-3" class="slide level2">
<h2>ATAC-seq</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/atac_haibo-liu.webp" class="" width="1000"></p>
<figcaption>Haibo Liu</figcaption>
</figure>
</div>
<!------------------------------------------------------------------>
</section>
<section id="atac-seq---footprinting" class="slide level2">
<h2>ATAC-seq - Footprinting</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/atac-footprinting_jeffrey-granja.png" class="" width="500"></p>
<figcaption>Jeffrey Granja</figcaption>
</figure>
</div>
<!------------------------------------------------------------------>
</section>
<section id="chip-seq---chromatin-immunoprecipitation-sequencing" class="slide level2">
<h2><span style="font-size: .75em;">ChIP-seq - Chromatin Immunoprecipitation Sequencing</span></h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/chip-protocol.png" class="quarto-figure quarto-figure-center" width="600"></p>
</figure>
</div>
<!------------------------------------------------------------------>
</section>
<section id="cutrun---cleavage-under-targets-and-release-using-nuclease" class="slide level2" style="font-size: .65em;">
<h2><span class="smaller">CUT&amp;RUN - Cleavage under targets and release using nuclease</span></h2>
<div class="columns">
<div class="column" style="width:60%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/cut-and-run-workflow.png" class="quarto-figure quarto-figure-center" width="450"></p>
</figure>
</div>
</div><div class="column" style="width:40%;">
<ul>
<li>Requires less starting material</li>
<li>Faster protocol</li>
<li>No cross-linking step
<ul>
<li>Decreases false positives</li>
<li>Less sensitive</li>
</ul></li>
</ul>
</div></div>
<!------------------------------------------------------------------>
</section>
<section id="cuttag---cleavage-under-targets-and-tagmentation" class="slide level2" style="font-size: .65em;">
<h2><span class="smaller">CUT&amp;TAG - Cleavage under targets and tagmentation</span></h2>
<div class="columns">
<div class="column" style="width:60%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/cut-and-tag-workflow.png" class="quarto-figure quarto-figure-center" width="450"></p>
</figure>
</div>
</div><div class="column" style="width:40%;">
<ul>
<li>Fastest</li>
<li>May not work on some targets</li>
</ul>
</div></div>
</section>
<section id="peaks" class="slide level2 smaller">
<h2>Peaks</h2>
<ul>
<li>These methods result in enrichment of DNA from specific genome regions</li>
<li>These enriched regions are called “peaks”</li>
<li>Peaks correspond to regions of interest
<ul>
<li>Transcription factor binding sites / histone modifications</li>
<li>Open chromatin regions</li>
</ul></li>
<li>Most software tools for identifying these peaks work with any data source
<ul>
<li>Parameters specific to one assay may be needed</li>
</ul></li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/atac-peaks_feng-yan.png" class="" width="700"></p>
<figcaption>Feng Yan</figcaption>
</figure>
</div>
<!------------------------------------------------------------------>
</section>
<section id="shared-analytical-workflow" class="slide level2">
<h2>Shared Analytical Workflow</h2>
<ol type="1">
<li>Quality Control</li>
<li>Read Trimming and Filtering</li>
<li>Read Alignment</li>
<li>Peak Calling</li>
<li>Downstream Analysis</li>
</ol>
<!------------------------------------------------------------------>
</section>
<section id="dataset" class="slide level2 smaller">
<h2>Dataset</h2>
<ul>
<li><a href="https://doi.org/10.1093/nar/gkab448">Lindeman et al.&nbsp;2021</a>
<ul>
<li>The conserved sex regulator DMRT1 recruits SOX9 in sexual cell fate reprogramming</li>
</ul></li>
<li>ATAC-seq and ChIP-seq data from mouse embryonic gonad tissue</li>
<li>Gonad differentiation involves major chromatin remodeling</li>
<li>DMRT1 is key transcription factor in remodelling</li>
<li>Enables binding of SOX9</li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="logging-in-to-the-hpc" class="slide level2">
<h2>Logging in to the HPC</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href=""></a><span class="fu">ssh</span> cbcusr<span class="op">&lt;</span>number<span class="op">&gt;</span>@xanadu-submit-ext.cam.uchc.edu</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<!------------------------------------------------------------------>
</section>
<section id="git-repository" class="slide level2">
<h2>Git Repository</h2>
<ul>
<li><p>All of the code and data for this workshop is available on GitHub.</p></li>
<li><p>You can clone the repository to your home directory on the HPC using the following command:</p></li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href=""></a><span class="fu">git</span> clone https://github.com/CBC-UCONN/ATAC-seq-ChIP-seq-Workshop</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="data-download" class="slide level2">
<h2>Data Download</h2>
<ul>
<li>NCBI Sequence Read Archive (SRA)</li>
<li><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA646465/">https://www.ncbi.nlm.nih.gov/bioproject/PRJNA646465/</a></li>
<li>Use SRA Toolkit to download NCBI SRA data</li>
<li>See:
<ul>
<li><code>dmrt/atac-seq/scripts/00_download.sh</code></li>
<li><code>dmrt/chip-seq/scripts/00_download.sh</code></li>
</ul></li>
<li>Obtained metadata from run selector</li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="fastq-format" class="slide level2 smaller">
<h2>FASTQ Format</h2>
<ul>
<li>Typical raw data format for sequencing reads</li>
<li>Typically separated into two files
<ul>
<li>One for each end of a read, if paired</li>
</ul></li>
<li>Each read is represented by four lines:
<ol type="1">
<li>Header (starts with ‘@’)</li>
<li>Sequence (the actual nucleotide sequence)</li>
<li>Separator (starts with ‘+’)</li>
<li>Quality score (ASCII-encoded quality scores for each nucleotide)</li>
</ol></li>
<li>Example:</li>
</ul>
<pre><code>@SEQUENCE_ID
GATTTGGGGTTTAAAGGGTGACCTGGTAGG
+
!''*((((***+))%%%++)(****)())-+**--1</code></pre>
<!------------------------------------------------------------------>
</section>
<section id="fastq-format---quality-scores" class="slide level2 smaller">
<h2>FASTQ Format - Quality Scores</h2>
<ul>
<li>Quality scores are encoded using ASCII characters</li>
<li>Each character represents a 2 digit quality score for the corresponding nucleotide
<ul>
<li>This permits more efficient storage</li>
</ul></li>
<li>The quality score is a measure of the confidence in the accuracy of the base call</li>
<li>The higher the quality score, the more reliable the base call</li>
<li>Calculated as <code>-10 * log10(P)</code> where <code>P</code> is the probability of an incorrect base call</li>
<li>Examples:
<ul>
<li>Q = 10, P = 0.1 (10% chance of error)</li>
<li>Q = 30, P = 0.001 (0.1% chance of error)</li>
</ul></li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="fastq-format---headers" class="slide level2 smaller">
<h2>FASTQ Format - Headers</h2>
<ul>
<li>The header line contains metadata about the read</li>
<li>Illumina headers typicallly look like this:</li>
</ul>
<pre><code>@HWUSI-EAS100R:6:73:941:1973#0/1</code></pre>
<p>The components of the header are:</p>
<ul>
<li>HWUSI-EAS100R: the unique instrument name</li>
<li>6: flowcell lane</li>
<li>73: tile number within the flowcell lane</li>
<li>941: ‘x’-coordinate of the cluster within the tile</li>
<li>1973: ‘y’-coordinate of the cluster within the tile</li>
<li>#0: index number for a multiplexed sample (0 for no indexing)</li>
<li>/1: the member of a pair, /1 or /2</li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="quality-control" class="slide level2">
<h2>Quality Control</h2>
<ul>
<li>Quality of reads</li>
<li>Quantify of reads</li>
<li>Adapter Contamination</li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="adapter-contamination" class="slide level2">
<h2>Adapter Contamination</h2>
<p>Can occur if insert size is too short</p>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/sequencing-by-synthesis.jpg" class="nostretch" width="350"></p>
<figcaption>Illumina Inc.</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/adapter_contamination-ecseq.png" class="nostretch" width="300"></p>
<figcaption>ECSEQ</figcaption>
</figure>
</div>
<!------------------------------------------------------------------>
</section>
<section id="fastqc" class="slide level2">
<h2>Fastqc</h2>
<ul>
<li>Evaluates several metrics for a FASTQ file</li>
<li>Provides a summary of the quality of the reads</li>
<li>Generates an HTML report with visualizations and statistics</li>
<li>Can be run with the command:</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/fastqc.jpg" class="" width="300"></p>
<figcaption>Babraham Institute</figcaption>
</figure>
</div>
<!------------------------------------------------------------------>
</section>
<section id="multiqc" class="slide level2">
<h2>Multiqc</h2>
<ul>
<li>Consolidates multiple reports from FastQC and other tools</li>
<li>Generates a single HTML report with visualizations and statistics</li>
<li>Useful for comparing multiple samples or conditions</li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="filtering-and-trimming" class="slide level2">
<h2>Filtering and Trimming</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>Remove low-quality reads</li>
<li>Remove adapter sequences</li>
<li>Many tools:
<ul>
<li>Trimmomatic</li>
<li>Trim Galore</li>
<li><strong>Fastp</strong></li>
<li>Atria*</li>
</ul></li>
</ul>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/illumina-read.jpg" class="quarto-figure quarto-figure-center" width="500"></p>
</figure>
</div>
</div></div>
<!------------------------------------------------------------------>
</section>
<section id="mapping-alignment" class="slide level2">
<h2>Mapping &amp; Alignment</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>Determine where each read originated from in the genome</li>
<li>Align reads to a reference genome</li>
<li>Common tools:
<ul>
<li>Bowtie2</li>
<li>STAR</li>
<li><strong>BWA</strong></li>
</ul></li>
</ul>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/read-mapping.png" class="quarto-figure quarto-figure-center" width="400"></p>
</figure>
</div>
</div></div>
<!------------------------------------------------------------------>
</section>
<section id="mapping-alignment---reference-genome" class="slide level2" style="font-size: .65em;">
<h2><span class="smaller">Mapping &amp; Alignment - Reference Genome</span></h2>
<ul>
<li>Representative sequence of a species’ genome
<ul>
<li>Arguably a poor representation due to genetic variation</li>
</ul></li>
<li>Genome Reference Consortium (GRC)
<ul>
<li>Produces high-quality reference genome assemblies</li>
<li>Human: GRCh38 (hg38)</li>
<li>Mouse: GRCm39 (mm39)</li>
</ul></li>
<li>Available from:
<ul>
<li><a href="https://genome.ucsc.edu/">UCSC Genome Browser</a></li>
<li><a href="https://ensemblgenomes.org/">Ensembl</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/genome/">NCBI</a></li>
</ul></li>
<li>Providers use slightly different naming conventions for chromosomes
<ul>
<li>Important to be consistent throughout analysis</li>
</ul></li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="reference-genome-index" class="slide level2">
<h2>Reference Genome Index</h2>
<ul>
<li>Data structure for quickly finding short nucleotide sequences in a large reference genome</li>
<li>Two main types used in read mapping:
<ul>
<li>FM Index (BWA, Bowtie2)</li>
<li>Suffix Array (STAR)</li>
</ul></li>
<li>Must be built before mapping reads</li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="sambam-file-formats" class="slide level2 smaller">
<h2>SAM/Bam file formats</h2>
<ul>
<li>Output is a SAM file</li>
<li>SAM (Sequence Alignment/Map) is a text format for storing aligned reads</li>
<li>It contains information about the alignment of each read to the reference genome</li>
<li>SAM files are human-readable, but can be large</li>
<li>They can be compressed to BAM format for storage and efficiency
<ul>
<li>They are compressed and indexed for efficient access</li>
<li>Not human-readable</li>
</ul></li>
<li>SAM/BAM files can be viewed with tools like IGV or samtools</li>
<li>They are used for downstream analyses like peak calling</li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="sambam-file-formats-1" class="slide level2 smaller">
<h2>SAM/Bam file formats</h2>
<ul>
<li>Example SAM file:</li>
</ul>
<pre><code>@HD VN:1.0  SO:unsorted
@SQ SN:chr1 LN:248956422
@RG ID:group1   SM:sample1
@PG ID:program1 PN:STAR VN:2.7.9a
read1   0   chr1    100 60  10M1I5M2D3M *   0   0   ACGTACGTAC  IIIIIIIIIIIIIIIIIIII    AS:i:10 NM:i:1</code></pre>
<p><br></p>
<div style="font-size: 0.6em;">
<table class="caption-top">
<thead>
<tr class="header">
<th>Col</th>
<th>Field</th>
<th>Type</th>
<th>Brief description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>QNAME</td>
<td>String</td>
<td>Query template NAME</td>
</tr>
<tr class="even">
<td>2</td>
<td>FLAG</td>
<td>Int</td>
<td>bitwise FLAG</td>
</tr>
<tr class="odd">
<td>3</td>
<td>RNAME</td>
<td>String</td>
<td>References sequence NAME</td>
</tr>
<tr class="even">
<td>4</td>
<td>POS</td>
<td>Int</td>
<td>1-based leftmost mapping POSition</td>
</tr>
<tr class="odd">
<td>5</td>
<td>MAPQ</td>
<td>Int</td>
<td>MAPping Quality</td>
</tr>
<tr class="even">
<td>6</td>
<td>CIGAR</td>
<td>String</td>
<td>CIGAR string</td>
</tr>
<tr class="odd">
<td>7</td>
<td>RNEXT</td>
<td>String</td>
<td>Ref. name of the mate/next read</td>
</tr>
<tr class="even">
<td>8</td>
<td>PNEXT</td>
<td>Int</td>
<td>Position of the mate/next read</td>
</tr>
<tr class="odd">
<td>9</td>
<td>TLEN</td>
<td>Int</td>
<td>observed Template LENgth</td>
</tr>
<tr class="even">
<td>10</td>
<td>SEQ</td>
<td>String</td>
<td>segment SEQuence</td>
</tr>
<tr class="odd">
<td>11</td>
<td>QUAL</td>
<td>String</td>
<td>ASCII of Phred-scaled base QUALity+33</td>
</tr>
</tbody>
</table>
</div>
<!------------------------------------------------------------------>
</section>
<section id="cigar-string" class="slide level2 smaller">
<h2>CIGAR String</h2>
<ul>
<li>CIGAR (Compact Idiosyncratic Gapped Alignment Report) string is a part of the SAM/BAM format</li>
<li>It describes how the read aligns to the reference genome</li>
<li>It consists of a series of operations, each represented by a character and a number</li>
<li>Common operations include:
<ul>
<li>M: match or mismatch</li>
<li>I: insertion</li>
<li>D: deletion</li>
<li>N: skipped region (intron)</li>
<li>S: soft clipping (part of the read is not aligned)</li>
</ul></li>
<li>Example CIGAR string: <code>10M1I5M2D3M</code>
<ul>
<li>This means 10 matches, 1 insertion, 5 matches, 2 deletions, and 3 matches</li>
</ul></li>
</ul>
</section>
<section id="sambam-file-formats-2" class="slide level2 smaller">
<h2>SAM/Bam file formats</h2>
<p>Example SAM file:</p>
<pre><code>@HD VN:1.0  SO:unsorted
@SQ SN:chr1 LN:248956422
@RG ID:group1   SM:sample1
@PG ID:program1 PN:STAR VN:2.7.9a
read1   0   chr1    100 60  10M1I5M2D3M *   0   0   ACGTACGTAC  IIIIIIIIIIIIIIIIIIII    AS:i:10 NM:i:1</code></pre>
<p><br> Optional fields:</p>
<ul>
<li>AS:i:10 - Alignment Score</li>
<li>NM:i:1 - Number of Mismatches</li>
</ul>
<!------------------------------------------------------------------>
</section>
<section id="mapping-qc" class="slide level2">
<h2>Mapping QC</h2>
<ul>
<li>Assess quality of alignments</li>
<li>Common metrics:
<ul>
<li>Mapping rate</li>
<li>Coverage</li>
<li>Duplicate reads</li>
<li>Orientation (for paired-end reads)</li>
<li>Mapping quality</li>
</ul></li>
<li>Tools:
<ul>
<li>Qualimap</li>
<li>Picard Tools</li>
<li><strong>samtools</strong></li>
</ul></li>
</ul>

</section></section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">

</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="day1_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="day1_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="day1_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="day1_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="day1_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="day1_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="day1_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="day1_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="day1_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="day1_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
            const codeEl = trigger.previousElementSibling.cloneNode(true);
            for (const childEl of codeEl.children) {
              if (isCodeAnnotation(childEl)) {
                childEl.remove();
              }
            }
            return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp('/' + window.location.host + '/');
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>